### PR TITLE
Added RoundStartingEventArgs

### PR DESCRIPTION
**Pull request by ui2506**
Added an event that is called before OnRoundStarted and can disable the start of the round :P

### CODE REVIEW

**Comment by Mar1ki** [comment]
😀

**Comment by ui2506** [comment]
👋

пт, 1 нояб. 2024 г., 00:29 Mariki ***@***.***>:

File: Exiled.Events/EventArgs/Server/RoundStartingEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// <copyright file="RoundStartingEventArgs.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Server
+{
+    /// <summary>
+    /// Contains nothing.
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Server/RoundStartingEventArgs.cs]: 🤔, it does contain something.

File: Exiled.Events/EventArgs/Server/RoundStartingEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        /// <param name="isAllowed">
+        /// <inheritdoc cref="IsAllowed" />
+        /// </param>
+        public RoundStartingEventArgs(bool isAllowed = true)
+        {
+            IsAllowed = isAllowed;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the event can continue.
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Server/RoundStartingEventArgs.cs]: Gets or sets a value indicating whether the round can start or not.

File: Exiled.Events/Handlers/Server.cs [hunk 3]
```diff
@@ -103,6 +103,11 @@ public static class Server
         /// </summary>
         public static Event ReloadedPermissions { get; set; } = new();
 
+        /// <summary>
+        /// Invoked before round will started.
```
Comment (xNexusACS) [inline Exiled.Events/Handlers/Server.cs]: Invoked before the round starts.

**Review by xNexusACS — CHANGES_REQUESTED** [review]

**Review by xNexusACS — APPROVED** [review]

File: Exiled.Events/EventArgs/Server/RoundStartingEventArgs.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+// <copyright file="RoundStartingEventArgs.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Server
+{
+    /// <summary>
+    /// Contains isallowed or not.
```
Comment (xNexusACS) [inline Exiled.Events/EventArgs/Server/RoundStartingEventArgs.cs]: One last touch before merging.

Contains all information before the start of the round.

Everything should be documented correctly.

**Review by xNexusACS — CHANGES_REQUESTED** [review]

**Review by xNexusACS — APPROVED** [review]

**Pull request closed without merging**